lib/membership.js

```diff
@@ -38,23 +40,42 @@ const insertNewMembership = async function (personUri, roleUri, organizationUri)
 
   const organizationStatement = organizationUri ? `${sparqlEscapeUri(membershipUri)} org:organization ${sparqlEscapeUri(organizationUri)} .` : '';
 
+  let status = ACCESS_ALLOWED_STATUS_URI;
+  if (organizationUri) {
+    const result = await query(`
+    PREFIX adms: <http://www.w3.org/ns/adms#>
+
+    SELECT ?organizationStatus
+    WHERE {
+      GRAPH <${USERS_GRAPH}> {
+        ${sparqlEscapeUri(organizationUri)} adms:status ?organizationUri .
+      }
+    } LIMIT 1`);
+
+    if (result.results.bindings.length) {
+      status = result.results.bindings[0].organizationStatus.value;
+    }
+  }
+
   await update(`
     PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
     PREFIX dct: <http://purl.org/dc/terms/>
     PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX adms: <http://www.w3.org/ns/adms#>
 
     INSERT DATA {
       GRAPH <${USERS_GRAPH}> {
         ${sparqlEscapeUri(membershipUri)} a org:Membership ;
            mu:uuid ${sparqlEscapeString(id)} ;
            org:member ${sparqlEscapeUri(personUri)} ;
            org:role ${sparqlEscapeUri(roleUri)} ;
+           adms:status ${sparqlEscapeUri(status)} ;
            dct:created ${sparqlEscapeDateTime(now)} .
         ${organizationStatement}
       }
     }`);
```

Comment:
In a previous release organization was considered optional, but this no longer holds with the new features added. Hence, I think this query (including the `organizationStatement` that was already there) can be simplified to a single query `INSERT { ... } WHERE { ... }`

Reply:
Looking deeper, 2 queries will still be required because we need to return the value of status. But in either case, I would remove the logic of organization being `null`.